meerkat/model/tensormodel.py

```diff
@@ -0,0 +1,277 @@
+from __future__ import annotations
+
+import itertools
+from typing import Dict, List
+
+import cytoolz as tz
+import torch
+from tqdm import tqdm
+
+from meerkat import DataPanel
+from meerkat.columns.embedding_column import EmbeddingColumn
+from meerkat.columns.instances_column import InstancesColumn
+from meerkat.columns.prediction_column import ClassificationOutputColumn
+from meerkat.columns.segmentation_column import SegmentationOutputColumn
+from meerkat.columns.tensor_column import TensorColumn
+from meerkat.model.activation import ActivationOp
+from meerkat.model.model import Model
+
+
+class TensorModel(Model):
+    def __init__(
+        self,
+        # identifier: str,
+        model,
+        is_classifier: bool = None,
+        task: str = None,
+        device: str = None,
+    ):
+
+        if model is None:
+            raise ValueError(
+                f"A PyTorch model is required with {self.__class__.__name__}."
+            )
+        super(TensorModel, self).__init__(
+            # identifier=identifier,
+            model=model,
+            is_classifier=is_classifier,
+            task=task,
+            device=device,
+        )
+
+        # Move the model to device
+        self.to(self.device)
+
+    def forward(self, input_batch: Dict) -> Dict:
+        # Run the model on the input_batch
+        with torch.no_grad():
+            outputs = self.model(input_batch)
+
+        if self.is_classifier:
+            # probs and preds can be handled at ClassificationOutputColumn
+            # TODO(Priya): See if there is any case where these are to be returned
+            output_dict = {"logits": outputs.to("cpu")}
+
+        elif self.task == "semantic_segmentation":
+            # Output is a dict with key 'out'
+            output_dict = {"logits": outputs["out"].to("cpu")}
+
+        elif self.task == "timeseries":
+            output_dict = {"preds": outputs.to("cpu")}
+
+        elif self.task == "instance_segmentation":
+            output_dict = {
+                "preds": [output["instances"].to("cpu") for output in outputs]
+            }
+
+        return output_dict
+
+    def process_batch(self, batch: DataPanel, input_columns: List[str]):
+
+        # Convert the batch to torch.Tensor and move to device
+        if self.task == "instance_segmentation":
+            input_batch = batch[input_columns[0]].data
+        else:
+            input_batch = batch[input_columns[0]].data.to(self.device)
+
+        return input_batch
+
+    def predict_batch(
+        self, dataset: DataPanel, input_columns: List[str], batch_size=32
+    ):
+        # Gathers predictions for entire dataset
+
+        predictions = []
```

Comment:
This can be done with an update and can be moved up to the `model` base class since the implementation is shared with huggingfacemodel

Reply:
A common _predict method is now present in `Model` class